File: data/multilang_comparison/program_cpp_array/src/path.cpp
#include "path.hpp"
#include <cstdlib>
#include <string>
#include <random>


unsigned long get_path() {
   const char* path = std::getenv("BENCH_PATH");
   if(path != nullptr) { 
      return std::stoul(path);
   }else {
      static std::mt19937_64 rng(std::random_device{}());
      return rng();
   }
}

File: data/multilang_comparison/program_nim/src/path.nim
proc getPath*(): uint64 =
   let path = getEnv("BENCH_PATH")
   if path.len > 0:
      return uint64(parseInt(path))
   else:
      let hi = uint64(benchgenRand())
      let lo = uint64(benchgenRand())
      return (hi shl 32) or lo

